grobidTagger.py

```diff
@@ -294,16 +291,21 @@ class Tasks:
             someTexts.append("Wilcoxon signed-ranks tests were performed to calculate statistical significance of comparisons between  alignment programs, which include ProbCons (version 1.10) (23), MAFFT (version 5.667) (11) with several options, MUSCLE (version 3.52) (10) and ClustalW (version 1.83) (7).")
             someTexts.append("The statistical analysis was performed using IBM SPSS Statistics v. 20 (SPSS Inc, 2003, Chicago, USA).")
 
-        result = annotate_text(someTexts, model, "json", use_ELMo=use_ELMo, architecture=architecture)
-        print(json.dumps(result, sort_keys=False, indent=4, ensure_ascii=False))
+        if architecture.find("FEATURE") == -1:
```

Comment:
Personally to me it seems slightly wrong to pass around the architecture to that many places. That should only be relevant for loading the model, or actually just the path to the model.

Here I would say, one shouldn't check the name of the architecture. The model or its config should provide the information whether it requires features. (In my case I am using a custom model that doesn't have `FEATURES` in it, but supports features, although doesn't require them)

Reply:
Yes you're right it should be a property of the model, it was a quick fix and I can re-work that.

Reply:
In fact looking at it again, here there is no simple way to control the usage of features, we only have the architecture name, no model, no config yet and I think we don't want to load/create a model just for this.

Reply:
True, it is more difficult in that case. I think I came across something similar related to the preprocessor. Currently the dependencies don't seem to be "clean". I believe the model is changing the preprocessor.
Both of those issue could probably be resolved by having a model factory or similar. Then you wouldn't actually load the model, but you could load the factory or a descriptor, which would tell you what is required.
Or in this particular case you could also just remove the condition here and let it fail later when the model is loaded. It could still have the same helpful error message which for me would be the the most important part.